{"name": "test_changjing_02", "status": "failed", "statusDetails": {"message": "AssertionError", "trace": "self = <test_changjing.TestChangJing object at 0x0000018D315994C8>\n\n    @allure.story('场景测试02')\n    def test_changjing_02(self):\n        ru = RequestUtil()\n        cjru = ChangJingRequestUtil('changjing_02', ru)\n>       cjru.yamlAndRequest()\n\n..\\testcases\\test_changjing.py:19: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\util\\changJingRequestUtil.py:66: in yamlAndRequest\n    self.requestUtil.assertResult(result, dict['validate']['statusCode'], dict['validate']['errorCode'],dict['validate']['msg'])\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <util.requestUtil.RequestUtil object at 0x0000018D315CE788>\nresult = <Response [200]>, statusCode = 200, errorCode = 400, msg = '用户名已存在'\n\n    def assertResult(self, result, statusCode, errorCode, msg):\n        with allure.step('发送请求'):\n            allure.attach(str(result.request.url),'请求路径')\n            allure.attach(str(result.request.method),'请求方式')\n            allure.attach(str(result.request.headers),'请求头')\n            allure.attach(str(result.request.body),'请求体')\n    \n        with allure.step('获取响应'):\n            allure.attach(str(result.status_code),'响应状态码')\n            allure.attach(str(result.headers),'响应头')\n            allure.attach(str(result.text),'响应文本内容')\n    \n        with allure.step('断言'):\n            allure.attach(str(statusCode),'预期响应状态码')\n            allure.attach(str(errorCode),'预期返回错误码')\n            allure.attach(str(msg),'预期返回错误信息')\n    \n            assert result.status_code == int(statusCode)\n>           assert result.json()['status'] == int(errorCode)\nE           AssertionError\n\n..\\util\\requestUtil.py:61: AssertionError"}, "steps": [{"name": "发送请求:mcheckImg", "status": "passed", "steps": [{"name": "发送请求", "status": "passed", "attachments": [{"name": "请求路径", "source": "f5f88b8c-ec42-4ad0-9319-4442b1319626-attachment.attach"}, {"name": "请求方式", "source": "4e96a41e-80a7-4211-a8aa-8698100c7843-attachment.attach"}, {"name": "请求头", "source": "d1184669-8cd5-45ca-b425-af214df4a1e3-attachment.attach"}, {"name": "请求体", "source": "403a36a8-57e4-4e57-80b2-0df8d222b9eb-attachment.attach"}], "start": 1610270379837, "stop": 1610270379839}, {"name": "获取响应", "status": "passed", "attachments": [{"name": "响应状态码", "source": "86bce98d-6ef7-437a-8f15-c1c522e22ce6-attachment.attach"}, {"name": "响应头", "source": "f8e37e3c-6e18-4424-b9c8-3fdbf9b57fd2-attachment.attach"}, {"name": "响应文本内容", "source": "4f128f77-c552-44bc-bdcd-65506293a4de-attachment.attach"}], "start": 1610270379839, "stop": 1610270379841}, {"name": "断言", "status": "passed", "attachments": [{"name": "预期响应状态码", "source": "aad3725f-98d0-4ba9-b620-391cb5b975e0-attachment.attach"}, {"name": "预期返回错误码", "source": "775f740c-b68c-4beb-bc1c-2abe3c701e63-attachment.attach"}, {"name": "预期返回错误信息", "source": "14c198cb-4957-43fb-bae4-ef73d3c2ca2b-attachment.attach"}], "start": 1610270379841, "stop": 1610270379843}], "start": 1610270379825, "stop": 1610270379863}, {"name": "发送请求:mregist", "status": "failed", "statusDetails": {"message": "AssertionError\n", "trace": "  File \"D:\\PycharmProjects\\ApiAutoTestPytest\\util\\changJingRequestUtil.py\", line 66, in yamlAndRequest\n    self.requestUtil.assertResult(result, dict['validate']['statusCode'], dict['validate']['errorCode'],dict['validate']['msg'])\n  File \"D:\\PycharmProjects\\ApiAutoTestPytest\\util\\requestUtil.py\", line 61, in assertResult\n    assert result.json()['status'] == int(errorCode)\n"}, "steps": [{"name": "发送请求", "status": "passed", "attachments": [{"name": "请求路径", "source": "fbeeeb7b-f640-453f-8b08-830534d16a32-attachment.attach"}, {"name": "请求方式", "source": "49127df3-f47e-47ef-abc3-9a551d686115-attachment.attach"}, {"name": "请求头", "source": "b707b138-86e2-4efb-b30a-1c5ac8fd5f03-attachment.attach"}, {"name": "请求体", "source": "eb1b214c-9da2-4b81-853e-61d5abfc1fda-attachment.attach"}], "start": 1610270379883, "stop": 1610270379885}, {"name": "获取响应", "status": "passed", "attachments": [{"name": "响应状态码", "source": "97b997f4-29d9-4c7b-9fb4-93d923129adb-attachment.attach"}, {"name": "响应头", "source": "1794c2f0-ec03-4a1c-8ae2-cafac0cb64de-attachment.attach"}, {"name": "响应文本内容", "source": "bfed91cd-26e3-482b-afb2-66304bed6036-attachment.attach"}], "start": 1610270379885, "stop": 1610270379886}, {"name": "断言", "status": "failed", "statusDetails": {"message": "AssertionError\n", "trace": "  File \"D:\\PycharmProjects\\ApiAutoTestPytest\\util\\requestUtil.py\", line 61, in assertResult\n    assert result.json()['status'] == int(errorCode)\n"}, "attachments": [{"name": "预期响应状态码", "source": "1bc82349-609c-4d20-bc6a-37a6bb32986a-attachment.attach"}, {"name": "预期返回错误码", "source": "de21de39-b738-4ff8-aa84-ea74b25ae864-attachment.attach"}, {"name": "预期返回错误信息", "source": "3db1da38-03b6-4d2b-a16a-b3c85d8193e0-attachment.attach"}], "start": 1610270379886, "stop": 1610270379889}], "start": 1610270379863, "stop": 1610270379890}], "start": 1610270379818, "stop": 1610270379890, "uuid": "fe7b33db-77ee-44ff-8df8-8d7b9e11d5a5", "historyId": "46532ac8e89544b76a25c2f1842b03ab", "testCaseId": "c53ec5dd6cfca030c1a5377767d78a45", "fullName": "testcases.test_changjing.TestChangJing#test_changjing_02", "labels": [{"name": "story", "value": "场景测试02"}, {"name": "feature", "value": "场景"}, {"name": "parentSuite", "value": "testcases"}, {"name": "suite", "value": "test_changjing"}, {"name": "subSuite", "value": "TestChangJing"}, {"name": "host", "value": "DESKTOP-HLLR26H"}, {"name": "thread", "value": "31024-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.test_changjing"}]}